flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/PhysicalTableSourceScan.scala

```diff
@@ -70,4 +73,18 @@ abstract class PhysicalTableSourceScan(
 
   def copy(traitSet: RelTraitSet, tableSource: TableSource[_]): PhysicalTableSourceScan
 
+  override def estimateRowCount(mq: RelMetadataQuery): Double = {
+    val tableSourceTable = getTable.unwrap(classOf[TableSourceTable[_]])
+
+    if (tableSourceTable.getStatistic != FlinkStatistic.UNKNOWN) {
```

Comment:
Right, but in the current implementation the statistics of the `TableSourceTable` are preferred over the stats of the `TableSource` (`if (tableSourceTable.getStatistic != FlinkStatistic.UNKNOWN) { return rowcount; }`).

If I see this correctly, the stats of the `TableSourceTable` are the stats in the catalog, but a `FilterableTableSource` knows if a filter has been pushed down and should have more accurate stats because it can apply the selectivity of the pushed down filter. 

So in my opinion the `TableSource` statistics should be preferred over the static `TableSourceTable` stats.

Is my reasoning correct or did I overlook something @godfreyhe?

Thanks, Fabian